itest/litd_custom_channels_test.go

```diff
@@ -2745,6 +2745,160 @@ func testCustomChannelsLiquidityEdgeCasesGroup(ctx context.Context,
 	testCustomChannelsLiquidtyEdgeCasesCore(ctx, net, t, true)
 }
 
+// testCustomChannelsMultiRFQReceive tests that a node creating an invoice with
+// multiple RFQ quotes can actually guide the payer into using multiple private
+// taproot asset channels to pay the invoice.
+func testCustomChannelsMultiRFQReceive(ctx context.Context, net *NetworkHarness,
+	t *harnessTest) {
+
+	lndArgs := slices.Clone(lndArgsTemplate)
+	litdArgs := slices.Clone(litdArgsTemplate)
+
+	// Explicitly set the proof courier as Zane (now has no other role
+	// other than proof shuffling), otherwise a hashmail courier will be
+	// used. For the funding transaction, we're just posting it and don't
+	// expect a true receiver.
+	zane, err := net.NewNode(
+		t.t, "Zane", lndArgs, false, true, litdArgs...,
+	)
+	require.NoError(t.t, err)
+
+	litdArgs = append(litdArgs, fmt.Sprintf(
+		"--taproot-assets.proofcourieraddr=%s://%s",
+		proof.UniverseRpcCourierType, zane.Cfg.LitAddr(),
+	))
+
+	charlie, err := net.NewNode(
+		t.t, "Charlie", lndArgs, false, true, litdArgs...,
+	)
+	require.NoError(t.t, err)
+
+	dave, err := net.NewNode(t.t, "Dave", lndArgs, false, true, litdArgs...)
+	require.NoError(t.t, err)
+	erin, err := net.NewNode(t.t, "Erin", lndArgs, false, true, litdArgs...)
+	require.NoError(t.t, err)
+	fabia, err := net.NewNode(
+		t.t, "Fabia", lndArgs, false, true, litdArgs...,
+	)
+	require.NoError(t.t, err)
+	yara, err := net.NewNode(
+		t.t, "Yara", lndArgs, false, true, litdArgs...,
+	)
+	require.NoError(t.t, err)
+
+	nodes := []*HarnessNode{charlie, dave, erin, fabia, yara}
+	connectAllNodes(t.t, net, nodes)
+	fundAllNodes(t.t, net, nodes)
+
+	// The topology we are going for looks like the following:
```

Comment:
Ah, so the helper function creates the right side of this topology and the left side is created here. Any reason to not also move the sats channel creation into the helper function? Seems like the topology created by the helper function otherwise isn't really complete and useful on its own?